### PR TITLE
Add utility function to rollback D&B updates

### PR DESCRIPTION
**Pull request by alixedi**
### Description of change

Add a utility function for rolling back DNB company updates for a given company.

I would recommend reviewing each commit separately.

### Checklist

* [ ] Has a new newsfragment been created? Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions
* [ ] Do any added or updated endpoints appear in the API documentation? See [docs/Maintaining the API documentation.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/Maintaining&#32;the&#32;API&#32;documentation.md) for more details
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/dnb_api/test/test_utils.py [hunk 1]
```diff
@@ -554,3 +558,74 @@ def test_get_company_dnb_service_request_error(
...
+        """
+        company = CompanyFactory(duns_number=formatted_dnb_company['duns_number'])
+
+        update_company_from_dnb(
+            company,
+            formatted_dnb_company,
+            update_descriptor='foo',
+        )
+
+        with pytest.raises(RevisionNotFoundError) as e:
```
Comment (reupen) [inline datahub/dnb_api/test/test_utils.py]: [Would suggest `excinfo` instead of `e` (it's an instance of ExceptionInfo)}(https://docs.pytest.org/en/latest/assert.html?highlight=raises#assertions-about-expected-exceptions):

```suggestion
        with pytest.raises(RevisionNotFoundError) as excinfo:
```

File: datahub/dnb_api/utils.py [hunk 2]
```diff
@@ -319,3 +327,42 @@ def get_company_update_page(last_updated_after, next_page=None):
...
+):
+    """
+    Given a company, an update descriptor that identifies a particular update
+    patch and fields that default to ALL_DNB_UPDATE_FIELDS, rollback the record
+    to the state before the update was applied.
+    """
+    fields_to_update = fields_to_update or ALL_DNB_UPDATED_MODEL_FIELDS
+    update_comment = f'Updated from D&B [{update_descriptor}]'
+    rollback_version = _get_rollback_version(company, update_comment)
+    fields = rollback_version.field_dict if fields_to_update is None else {
```
Comment (reupen) [inline datahub/dnb_api/utils.py]: Can `fields_to_update` be `None` here?

Comment (alixedi) [reply datahub/dnb_api/utils.py]: Relic.

File: datahub/dnb_api/test/test_utils.py [hunk 3]
```diff
@@ -554,3 +558,74 @@ def test_get_company_dnb_service_request_error(
...
+        """
+        Test that rollback_dnb_company_update will fail with the given error
+        message when there is an issue in finding the version to revert to.
+        """
+        company = CompanyFactory(duns_number=formatted_dnb_company['duns_number'])
+
+        update_company_from_dnb(
+            company,
+            formatted_dnb_company,
+            update_descriptor='foo',
```
Comment (mcbhenwood) [inline datahub/dnb_api/test/test_utils.py]: Is the success of this test predicated upon the idea that (unlike in `test_rollback`) you don't initiate Reversion management on the original Company record? So then 'foo' is the base version because the `update_company_from_dnb` just acts on a versionless object?

If so then it feels a bit like this test is based too heavily on knowing how the implmentation works. Unless I am misunderstanding which is highly likely. Maybe this test is for a particular edge case I don't know about.

Comment (alixedi) [reply datahub/dnb_api/test/test_utils.py]: This test is meant to tease out a couple of errors. One of those errors happens when there is no version before the update which is why I chose to not have a base version for `company`.

This is an unlikely corner case which is not easy to recover from so this test is essentially checking whether we are logging this properly.
